WebApp/autoreduce_webapp/selenium_tests/tests/test_help_page.py

```diff
@@ -20,4 +20,5 @@ def test_all_sidebar_contents_generated(self):
         Test that the number of elements in the sidebar equals the number of
         topics on the page
         """
-        WebDriverWait(self.driver, 10).until(len(self.page.get_sidenav_contents()) == len(self.page.get_help_topics()))
+        WebDriverWait(self.driver,
```

Comment:
This will still pass even if no help topics or contents exist on the page

Reply:
I'm on the fence about adding `== constant` because every time someone want to add another topic they will have to change the test. You shouldn't have to change the tests for something that is allowed.

I've added a test that checks that `topics>0` because I agree a Help page should logically include help info. But I'm interested to hear which way you think is _good practice_ for unit tests.

Reply:
asserting > 0 does not mean they are all loaded. If the test changes dynamically it is not deterministic and is only testing that at least 1 condition hasn't gone wrong and not that everything is working as should.

E.g. if lets say 12 topics should load, but because something has been broken only the first 6 load, the test passes and we wont know there is an issue unless we notice or it is reported to us.

 Increasing the number in one line is also a very small overhead

Reply:
As the help is currently hard coded in the HTML file I think we can be confident that they are all rendered. 

It's useful to have both `topics > 0` and `len(self.page.get_sidenav_contents()) == len(self.page.get_help_topics())` checks IMO (maybe in the same test), as that will cause a failure if nothing is rendered (e.g. an error in the view that causes a 500 will render nothing).

I don't think we need to check for the exact number of topics shown until we are adding them dynamically (e.g. via a django templated for loop)

Reply:
The sidebar is generated dynamically though I thought?